tests/ThreadClientTests.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NUnit.Framework;
+using StreamChat.Clients;
+using StreamChat.Models;
+
+namespace StreamChatTests
+{
+    [Parallelizable(ParallelScope.None)]
+    [TestFixture]
+    public class ThreadClientTests : TestBase
+    {
+        private ChannelWithConfig _channel;
+        private UserRequest _user;
+        private Message _parentMessage;
+        private Message _replyMessage;
+
+        [SetUp]
+        public async Task SetupAsync()
+        {
+            _user = await UpsertNewUserAsync();
+            _channel = await CreateChannelAsync(createdByUserId: _user.Id, members: new[] { _user.Id });
+
+            // Create a parent message to start a thread
```

Comment:
Comments are unnecessary if the code is straightforward. I would only reserve them for complex cases or when the intention is not clear. One downside of comments in method bodies is that they can easily get out of sync if someone updates the test in the future but forgets to rewrite the comment